deepcell/applications/application.py

```diff
@@ -312,6 +312,54 @@ def _resize_output(self, image, original_shape):
 
         return image
 
+    def _batch_predict(self,
+                       tiles,
+                       batch_size):
+        """Batch process tiles to generate model predictions.
+
+        The built-in keras.predict function has support for batching, but loads the entire
+        image stack into GPU memory, which is prohibitive for large images. This function uses
+        similar code to the underlying model.predict function without soaking up GPU memory.
+
+        Args:
+            tiles: Tiled data which will be fed to model
+            batch_size: Number of images to predict on per batch
+
+
+        Returns:
+            list: Model outputs
+        """
+
+        num_images = tiles.shape[0]
+        num_batches = int(np.ceil(num_images / float(batch_size)))
+        batches = [(i * batch_size, min(num_images, (i + 1) * batch_size)) for i in
+                   range(0, num_batches)]
+
+        # list to hold final output
+        output_tiles = []
+
+        # loop through each batch
+        for batch_index, (batch_start, batch_end) in enumerate(batches):
+            batch_inputs = tiles[batch_start:batch_end, ...]
+
+            batch_outputs = self.model.predict(batch_inputs, batch_size=batch_size)
+
+            # model with only a single output gets temporarily converted to a list
+            if type(batch_outputs) != list:
+                batch_outputs = [batch_outputs]
+
+            # initialize output list with empty arrays to hold all batches
+            if batch_index == 0:
+                for batch_out in batch_outputs:
+                    shape = (num_images,) + batch_out.shape[1:]
+                    output_tiles.append(np.zeros(shape, dtype=tiles.dtype))
```

Comment:
Since `output_tiles` is a list, we can just do a truthy-ness check to verify if it needs to be initialized:

```suggestion
            # initialize output list with empty arrays to hold all batches
            if not output_tiles:
                for batch_out in batch_outputs:
                    shape = (num_images,) + batch_out.shape[1:]
                    output_tiles.append(np.zeros(shape, dtype=tiles.dtype))
```